src/comm/LinkManager.cc

```diff
@@ -103,41 +103,39 @@ void LinkManager::createConnectedLink(LinkConfiguration* config)
 
 bool LinkManager::createConnectedLink(SharedLinkConfigurationPtr& config, bool isPX4Flow)
 {
-    LinkInterface* link = nullptr;
+    SharedLinkInterfacePtr link = nullptr;
 
     switch(config->type()) {
 #ifndef NO_SERIAL_LINK
     case LinkConfiguration::TypeSerial:
-        link = new SerialLink(config, isPX4Flow);
+        link = std::make_shared<SerialLink>(config, isPX4Flow);
```

Comment:
I assume this causes the constructors needing to be public. Kind of a bummer since nobody else should ever create these. But not the end of the world.  With the link constructors going public then I believe the friend class to LinkManager in all the Links can go away as well. Can you remove those. I'm 90% sure that's the only reason they are friended togehter.

Reply:
Yes that's the downside. You could work around it using factory methods, but that didn't really look appealing to me.
Indeed they can be removed.